wooting-macro-backend/src/plugin/system_event.rs

```diff
@@ -1,32 +1,23 @@
 use super::util;
+use anyhow::Result;
 use copypasta::{ClipboardContext, ClipboardProvider};
 use fastrand;
-use log::*;
 use rdev;
+use std::path::PathBuf;
 use std::vec;
 use tokio::sync::mpsc::UnboundedSender;
+use url::Url;
 
-use crate::hid_table::SCANCODE_TO_RDEV;
-
-// The Brightness of monitors is not implemented for macOS, so we have to condition the code.
-#[cfg(target_os = "windows")]
-use brightness::{windows::BrightnessExt, Brightness, BrightnessDevice};
-#[cfg(target_os = "linux")]
-use brightness::{Brightness, BrightnessDevice};
-
-#[cfg(any(target_os = "windows", target_os = "linux"))]
-use futures::{StreamExt, TryFutureExt};
-
-// Frequently used constants
+// Frequently used keys within the code.
 const COPY_HOTKEY: [rdev::Key; 2] = [rdev::Key::ControlLeft, rdev::Key::KeyC];
 const PASTE_HOTKEY: [rdev::Key; 2] = [rdev::Key::ControlLeft, rdev::Key::KeyV];
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, PartialEq, Hash, Eq)]
 #[serde(tag = "type")]
 pub enum DirectoryAction {
-    Directory { data: String },
-    File { data: String },
-    Website { data: String },
+    File { data: PathBuf },
+    Directory { data: PathBuf },
+    Website { data: Url },
```

Comment:
Changed to a much more representative datatypes to error correct the user's input at runtime by giving him a good error message feedback in the frontend